dask_expr/io/parquet.py

```diff
@@ -226,10 +227,11 @@ def _meta(self):
             return meta[column]
         return meta
 
-    @property
+    @cached_property
     def _plan(self):
         dataset_info = self._dataset_info
-        dataset_token = tokenize(dataset_info)
+        # Need to serialize dataset_info for deterministic tokenize result
+        dataset_token = tokenize(pickle.dumps(dataset_info))
```

Comment:
Can I ask about this.  What was non-deterministic here?  Is it that there is some arrow type here or something that isn't supportded by `normalize_token`?

Reply:
Yeah, I haven't been able to look into this yet. It may just be the `pyarrow.dataset.Dataset` object. If so, we can probably just find a way to register it with `normalize_token`.

Reply:
Okay, I was able to get deterministic tokenization by registering special `normalize_token` logic for `pyarrow.dataset.Dataset`, pyarrow.dataset.FileFormat, and `pyarrow.Schema`.